src/main/java/org/folio/dao/fund/FundPostgresDAO.java

```diff
@@ -49,6 +54,44 @@ public Future<List<Fund>> getFundsByIds(List<String> ids, DBConn conn) {
     return getFundsByCriterion(criterionBuilder.build(), conn);
   }
 
+  @Override
+  public Future<Boolean> isFundStatusChanged(Fund fund, DBConn conn) {
+    return getFundById(fund.getId(), conn)
+      .map(existingFund -> existingFund.getFundStatus() != fund.getFundStatus());
+  }
+
+  @Override
+  public Future<Void> updateRelatedCurrentFYBudgets(Fund fund, DBConn conn) {
+    String fullBudgetTableName = getFullTableName(conn.getTenantId(), BUDGET_TABLE);
+    String fullFYTableName = getFullTableName(conn.getTenantId(), FISCAL_YEAR_TABLE);
+
+    String sql = "UPDATE " + fullBudgetTableName + " SET jsonb = jsonb_set(jsonb,'{budgetStatus}', $1) " +
+      "WHERE((fundId=$2) " +
+      "AND (budget.fiscalYearId IN " +
+      "(SELECT id FROM " + fullFYTableName + " WHERE  current_date between (jsonb->>'periodStart')::timestamp " +
+      "AND (jsonb->>'periodEnd')::timestamp)));";
+
+    return conn.execute(sql, Tuple.of(fund.getFundStatus().value(), UUID.fromString(fund.getId())))
+      .mapEmpty();
+  }
+
+  @Override
+  public Future<Void> updateFund(Fund fund, DBConn conn) {
+    logger.debug("Trying to update finance storage fund by id {}", fund.getId());
+    return conn.update(FUND_TABLE, fund, fund.getId())
+      .onSuccess(x -> logger.info("Fund record {} was successfully updated", fund))
```

Comment:
fund does not have toString, basically logging just id is enough

Reply:
fixed